src/selectors/nodes.js

```diff
@@ -49,6 +53,18 @@ export const getNodeSelected = createSelector(
     )
 );
 
+/**
+ * Gets a map of nodeIds to nodes
+ */
+export const getNodesById = createSelector(
+  [state => state.graph.nodes],
+  (nodes = []) =>
+    nodes.reduce((result, node) => {
+      result[node.id] = node;
+      return result;
+    }, {})
+);
+
```

Comment:
tbh if this selector is only used for metadata you could put it in the metadata selector file if you like. This file's already pretty long and it's already starting to get broken up with some other selectors moving to other files to resolve dependency issues etc.

Reply:
I think since this is such a general selector, one that will likely be reused in future, it feels more natural for it to be in this layer?

Reply:
Fair enough, I think there's decent arguments either way, feel free to leave it